mockgcp/mockaiplatform/dataset.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockaiplatform
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	longrunning "google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/emptypb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/common/projects"
+	pb "github.com/GoogleCloudPlatform/k8s-config-connector/mockgcp/generated/mockgcp/cloud/aiplatform/v1beta1"
+)
+
+type datasetService struct {
+	*MockService
+	pb.UnimplementedDatasetServiceServer
+}
+
+func (s *datasetService) GetDataset(ctx context.Context, req *pb.GetDatasetRequest) (*pb.Dataset, error) {
+	name, err := s.parseDatasetName(req.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+
+	obj := &pb.Dataset{}
+	fmt.Printf("[debug] getting %s\n", fqn)
```

Comment:
Nit: probably better to `log := klog.FromContext(ctx)` and then `log.Info`, or just `klog.Infof` , but certainly not a blocker, just one for the future!

(We also have reasonable logging of the requests etc anyway in our tests, so you may not need your own logging.  But it might be that this is just left over for dev, I'm always doing that :-) )

Reply:
Sorry these are just something I randomly added during my debugging. I'll remove them.